src/lib/ext-activitylog.js

```diff
@@ -0,0 +1,143 @@
+import { save, load } from '../lib/save-load.js';
+
+export default class ActivityLog {
+  constructor() {
+    this.logs = [];
+    this.loadedLogs = [];
+    this.rowIndex = 0;
+
+    this.loadLogBtn = document.getElementById('loadLogBtn');
+    this.logInputFile = document.getElementById('logInputFile');
+    this.saveLogBtn = document.getElementById('saveLogBtn');
+    this.showLogDetailWrapper = document.querySelector(
+      '.show-log-detail-wrapper'
+    );
+    this.showLogDetails = document.getElementById('showLogDetails');
+    this.showLoadedLog = document.getElementById('showLoadedLog');
+    this.logTableWrapper = document.querySelector('.log-table');
+    this.table = document.querySelector('table');
+    this.notice = document.querySelector('.notice');
+    this.closeBtn = document.querySelector('.close');
+  }
+
+  async getExistingLogs() {
+    const { existingLogs } = await browser.runtime.sendMessage({
+      requestType: 'sendAllLogs',
+      requestTo: 'ext-monitor',
+    });
+    return existingLogs;
+  }
+
+  async loadFile(file) {
+    try {
+      const loadedLogsAsText = await load.loadLogAsText(file);
+      const loadedJSONLogs = JSON.parse(loadedLogsAsText);
+      this.loadedLogs.push(...loadedJSONLogs);
+      // TODO: Display loaded logs in table view
```

Comment:
Personally I think that we may defer it to a separate follow up where we can focus on it, and keep this one simpler and more focused on the logs table view.

Anyway, given that I did take a look to it, I guess that we shouldn't accumulate logs loaded from unrelated log files (at the moment I'm not even sure if this class should have both the `this.logs` and `this.loadedLogs` array, because it isn't clear who will be responsible to handle the loaded logs, where they will be rendered etc.).

Reply:
For now, I have taken off this part. I will send a follow up pr for load logs.